File: spec/javascript/engine/redis_spec.js
JS.ENV.Engine.RedisSpec = JS.Test.describe("Redis engine", function() { with(this) {
  before(function() {
    this.engineOpts = {type: Faye.Engine.Redis, password: "foobared", namespace: new Date().getTime().toString()}
  })
  after(function() { this.clean_redis_db() })
  
  itShouldBehaveLike("faye engine")
  
  describe("distribution", function() { with(this) {
    itShouldBehaveLike("distributed engine")
  }})
  
  describe("using a Unix socket", function() { with(this) {
    before(function() { with(this) {
      this.engineOpts.socket = "/tmp/redis.sock"
    }})
    
    itShouldBehaveLike("faye engine")
  }})
}})

File: spec/ruby/engine/redis_spec.rb
require "spec_helper"

describe Faye::Engine::Redis do
  let(:engine_opts)  { {:type => Faye::Engine::Redis, :password => "foobared", :namespace => Time.now.to_i.to_s} }
  
  after do
    engine.disconnect
    redis = EM::Hiredis::Client.connect('localhost', 6379)
    redis.auth(engine_opts[:password])
    redis.flushall
  end
  
  it_should_behave_like "faye engine"
  it_should_behave_like "distributed engine"
  
  describe "using a Unix socket" do
    before { engine_opts[:socket] = "/tmp/redis.sock" }
    it_should_behave_like "faye engine"
  end
end

